tasks/setup.yaml

```diff
@@ -1,6 +1,35 @@
+variables:
+  - name: REGISTRY1_TEST_IMAGE
+    default: "registry1.dso.mil/ironbank/opensource/defenseunicorns/zarf/zarf-agent"
+  - name: REGISTRY
+    default: "${REGISTRY}"
+  - name: REGISTRY_USERNAME
+    default: "${REGISTRY_USERNAME}"
+  - name: REGISTRY_PASSWORD
+    default: "${REGISTRY_PASSWORD}"
+  - name: REGISTRY_RETRY_INTERVAL
+    default: "${REGISTRY_RETRY_INTERVAL}"
+
 tasks:
   - name: k3d-test-cluster
     actions:
       - description: Create k3d cluster with UDS Core Istio
         # renovate: datasource=github-tags depName=defenseunicorns/uds-core versioning=semver
         cmd: uds deploy oci://defenseunicorns/uds/bundles/k3d-core-istio-dev:0.10.0-${UDS_ARCH} --confirm --no-progress
+
+  - name: registry-login
+    actions:
+      - cmd: |
+          echo "${REGISTRY_PASSWORD}" | uds zarf tools registry login \
+            --username "${REGISTRY_USERNAME}" \
+            --password-stdin \
+            "${REGISTRY}" \
+            >/dev/null
+
+          if [ "${REGISTRY}" = "registry1.dso.mil" ]; then
+             uds zarf tools registry digest \
+               "${REGISTRY1_TEST_IMAGE}" \
+               >/dev/null || (sleep "${REGISTRY_RETRY_INTERVAL}"; exit 1)
+          fi 
+        maxRetries: 10
```

Comment:
I honestly don't know:  do we need 10 retries here.  We could have almost a minute waiting on this to fail with a 5 second interval.  Could we fail sooner?  Maybe this is not a big deal in the long run but it could add up.

Reply:
@justin-o12 - this was a decision to implement a greater number of retries which can either be run more quickly or slowly by adjusting the retry interval.  at the moment there is no way to influence the number programmatically, so it is hardcoded until it can be parameterized.

Reply:
Just to add my 2 cents - think this is OK since the scenario we're catching here is almost always short lived with registry1 dropping connections. I'd rather have my pipeline retry for 60 seconds and pass at the last second vs having to manually trigger a retry in GH 😁